src/modules/water-demand-prediction/water-demand-prediction.component.ts

```diff
@@ -0,0 +1,658 @@
+        },
+        error: error => {
+          console.log(error);
+        },
+        complete: () => {
+          let newDataset = this.createNewDataset(
+            this.currentPredictedSmartmeterData()!.value,
+            this.currentPredictedSmartmeterData()!.name,
+            this.currentPredictedSmartmeterData()!.resolution,
+            this.currentPredictedSmartmeterData()!.timeframe,
+            false,
+            this.chartType(),
+          );
+
+          let realDataset = this.createNewDataset(
+            this.currentPredictedSmartmeterData()!.realValue,
+            "real values" + this.currentPredictedSmartmeterData()!.name,
+            this.currentPredictedSmartmeterData()!.resolution,
+            this.currentPredictedSmartmeterData()!.timeframe,
+            false,
+            this.chartType(),
+          );
+
+          let lower_conf_int = this.createNewDataset(
+            this.currentPredictedSmartmeterData()!.lower_conf_values,
+            "lower_confidence_interval",
+            this.currentPredictedSmartmeterData()!.resolution,
+            this.currentPredictedSmartmeterData()!.timeframe,
+            0,
+            this.chartType(),
+          );
+
+          let upper_conf_int = this.createNewDataset(
+            this.currentPredictedSmartmeterData()!.upper_conf_values,
+            "upper_confidence_interval",
+            this.currentPredictedSmartmeterData()!.resolution,
+            this.currentPredictedSmartmeterData()!.timeframe,
+            0,
+            this.chartType(),
+          );
+
+          let smartmeterdata: PredictedSmartmeterDataset = {
+            upper_conf_interval_dataset: upper_conf_int,
+            dataset: newDataset,
+            realValue_dataset: realDataset,
+            lower_conf_interval_dataset: lower_conf_int,
+            labels: this.currentPredictedSmartmeterData()!.date,
+          };
+
+          if (
+            !this.predictedDatasets[
+              this.currentPredictedSmartmeterData()!.resolution
+            ]
+          ) {
+            this.predictedDatasets[
+              this.currentPredictedSmartmeterData()!.resolution
+            ] = [];
+          }
+
+          this.predictedDatasets[
+            this.currentPredictedSmartmeterData()!.resolution
+          ].push(smartmeterdata);
+
+          this.setDisplayedResolution(
+            this.currentPredictedSmartmeterData()!.resolution,
+          );
+
+          this.showPredictedDatasets(
+            this.currentPredictedSmartmeterData()!.resolution,
+          );
+        },
+      });
+  }
+
+  /**
+   * checks first if all 3 choices are set and returns false when not.
+   * afterwards check if resolution key is already in datasets, true when not.
+   * last create colorcode from parameters and check if colorcode already present.
+   * when colorcode not present, request is unique and thus true.
+   * @param pred flag im request is for real or predicted data
```

Comment:
It might have more elegant to put your datasets in a record with two entries or in an array with two entries and enums as indexes.